cloudvolume/datasource/precomputed/mesh/sharded.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+
+from ..sharding import ShardingSpecification, ShardReader
+from ....storage import SimpleStorage
+from ....mesh import Mesh
+from ....lib import red
+
+class ShardedMultiLevelPrecomputedMeshSource:
+    def __init__(self, meta, cache, config, readonly=False):
+        self.meta = meta
+        self.cache = cache
+        self.config = config
+        self.readonly = bool(readonly)
+
+        spec = ShardingSpecification.from_dict(self.meta.info['sharding'])
+        self.reader = ShardReader(meta, cache, spec)
+
+        self.vertex_quantization_bits = self.meta.info['vertex_quantization_bits']
+        self.lod_scale_multiplier = self.meta.info['lod_scale_multiplier']
+        self.transform = np.array(self.meta.info['transform'] + [0,0,0,1]).reshape(4,4)
+    
+        if np.any(self.transform * np.array([[0,1,1,1],[1,0,1,1],[1,1,0,1],[1,1,1,0]])):
+            raise ValueError(red("Non-scale homogeneous transforms are not implemented"))
+
+    @property
+    def path(self):
+        return self.meta.mesh_path
+    
+    def get(self, segids, lods=None):
+        """Fetch meshes at all levels of details.
+
+        Parameters:
+        segids : int or [int ..]
+            Scalar or list of integer segment IDs.
+        lods : int, [int, ..] or None
+            Level of detail(s) to retrieve.  0 is highest level of detail.
+            None will fetch meshes at all levels.
+
+        Returns:
+        [Mesh, ..] or [[Mesh, ..], ..]
```

Comment:
In other Mesh sources, I believe we return `{ segid: Mesh or [ Meshes ], ... }`

Reply:
I may have overloaded this function here: scalar segid & scalar lod -> [ Mesh fragments], one being scalar one being a list -> [[Mesh fragments]], both being lists -> [[[Mesh fragments]]].  Eeek!

Reply:
Current return is now a simple  { segid: {lod: { Mesh, ... } } }.... I don't love it though.

Reply:
... and now it's { lod: { segid: Mesh } }.